py-polars/src/conversion.rs

```diff
@@ -599,7 +599,24 @@ fn materialize_list(ob: &PyAny) -> PyResult<Wrap<AnyValue>> {
 
 impl<'s> FromPyObject<'s> for Wrap<AnyValue<'s>> {
     fn extract(ob: &'s PyAny) -> PyResult<Self> {
-        let type_name = ob.get_type().name()?;
+        let mut type_name = ob.get_type().name()?;
```

Comment:
I notice this was already here, but I think we should put this later int he function as well so that we don't set this variable unneeded when converting primitives.